### ISSUE TITLE
Firebase same origin policy issue with Chrome 58+ in iOS 10.3.x+

### ISSUE THREAD

**Issue by fabiozaffani** [firebase/firebase-js-sdk#84]
### [REQUIRED] Describe your environment
  * Operating System version: iOS 10.3.x +
  * Firebase SDK version: 4.1.2
  * Firebase Product: auth or database (not sure which one is triggering the error)

### [REQUIRED] Describe the problem
The problem can be observed by using a Chrome broswer version 58+ in any iOS 10.3.+, the first time the issue was observed were in April 10. Basically i'm getting a console.error that is hapenning only in this specific environment. The error message is as follows:

Blocked a frame with origin "https://superpouco.com.br" from accessing a frame with origin "https://superpouco.firebaseapp.com". Protocols, domains, and ports must match.

Where "superpouco" is the domain i'm currently using, the issue also appears in another app under another domain that is using the same app with firebase.

I couldn't find anything related to this issue online so I wonder if it might be how I set the application or if it is due to the fact that it's hapenning in a very very specific scenario.

#### Steps to reproduce:
Basically accessing the app using Chrome version in the iOS described version.

**Comment by bojeil-google** [comment]
I don't have a 10.3 iOS device on me, but I tested with an iPhone 7 Plus running 10.2.1,  iPhone 6s Plus running 11.0 both with Chrome 59 and I was able to use signInWithPopup/Redirect which depend on the cross origin iframe, successfully. My test app has a different origin as my authDomain (iframe domain).

**Comment by fabiozaffani** [comment]
Hi @bojeil-google , thansk for the reply.

The error actually only starting triggering in version 10.3.1 of iOS, at least the oldest one I have logged here throught sentry.io was from April 10. The most recent one was yesterday and the iOS version was 10.3.2 from an iPAD device.

Also I think this might be related to the signInAnonymously since it will always trigger for new users arriving at my App.

An example of this most recent error timeline of events follows:

xhr POST https://www.googleapis.com/identitytoolkit/v3/relyingparty/signupNewUser?key=[redacted] [200] at **21:40:33**
xhr 	
POST https://www.googleapis.com/identitytoolkit/v3/relyingparty/getAccountInfo?key=[redacted] [200] at **21:40:35**
exception
SecurityError: Blocked a frame with origin "https://impallets.com.br" from accessing a frame with origin "https://impallets.firebaseapp.com". Protocols, domains, and ports must match. at **21:40:38**

**Comment by bojeil-google** [comment]
I doubt it is related to signInAnonymously as that operation doesn't require inter frame communication. The timing deceivingly makes it seem so. We embed an iframe https://impallets.firebaseapp.com/__/auth/iframe on load. We need to embed it early for certain reasons (long story) needed to make signInWithPopup work. When that iframe is embedded, it communicates to the parent page that it is ready. Not sure if this is related. Can you test calling signInWithPopup/signInWithRedirect and report what happens in that case.

As I mentioned I tested on a couple of devices and I was not able to replicate this. Don't have the exact device on me running 10.3.2, but will try to get hold of one in the coming week. In general I noticed the same version of chrome behaves the same on 10.x and 9.x.

**Comment by fabiozaffani** [comment]
@bojeil-google Sure bojeil, will keep watching the errors and try to extract any more information that i think might be helpfull. It's quite troublesome to debug since despite having an iOS6 with 10.3+ running on it, i'm currently with win10 so the debug doesn't work properly (itunes is a pain to try to sync iphone with win10).

It also looks like the error is non-blocking/fatal, since I can freely navigate around the site with my iphone, it triggers the error ( i can keep track of it in sentry ) but it doesn't disrupt the navigation or cause any other serious issue with the site.

Still, the errors keep comming daily and it is always the same config, ios 10.3.+ using chrome. My main worry is if this scalates in the next version, turning in a more serious issue with iOS 10.4.+. It's kind hard to predict.

**Comment by bojeil-google** [comment]
As I recall, I also tested on 11.0, the latest upcoming version of iOS and I couldn't replicate it.

**Comment by alphacentory** [comment]
I am running chrome 60+ and ios 10.3.3 I am getting this error, with a network error being thrown shortly after. I ran this on 10.3.1 literally a week ago and had no issues.

If it helps the stack is, 
Cordova based app using es6, nothing special what so ever. I have used this in simulators, I have used this on phones both android and ios and have no issues. 


I could really use some help.

**Comment by fabiozaffani** [comment]
@alphacentory That's good to know, for the past month I have been doing some code changes in my app but the error is still hapenning, even now after the iOS update to 10.3.3, so it's quite a relief to know that it may not be related to the app and actually be some issue with firebase js sdk.

**Comment by alphacentory** [comment]
@fabiozaffani I have no had anytime to dig further into this. Downside my project has a deadline of a couple of weeks and I still have no clue what this issue could be. 

Anything new on your end?

**Comment by fabiozaffani** [comment]
@alphacentory Still having the same issues, what's your current stack? Mine is:
React 15+
Immutable.js (currently removing it in favor of pure ES6 compiled through babel to ES5)
Redux (basic implementation with thunks)
Syncing redux data through firebase injecting data in the redux.

**Comment by fabiozaffani** [comment]
@bojeil-google hi bojeil, is there any news from your part about reproducing the issue or having some other reports about that?

**Comment by bojeil-google** [comment]
I am sorry but I can't reproduce this.

**Comment by alphacentory** [comment]
@fabiozaffani @bojeil-google I am going to have time coming up to further dig into this.(next week or so.) Hopefully someone digs into it before I do. But please leave this ticket open.

**Comment by jerryemmanuel** [comment]
Try adding CRF response header mentioning the iframe's url as a known url.

Content-Security-Policy: frames-ancestors 'self' https://www.childiframeurl.com

**Comment by alphacentory** [comment]
I am actually no longer getting this specific error.

@fabiozaffani how about you? 
I just started retesting this specific app and the previously mentioned issue, is no longer existent.

**Comment by jshcrowthe** [comment]
Closing due to inactivity. Feel free to re-open if there is more information!
